src/ncp/ncp_base_mtd.cpp

```diff
@@ -1136,8 +1134,7 @@ otError NcpBase::InsertPropertyHandler_THREAD_OFF_MESH_ROUTES(void)
 
     VerifyOrExit(mAllowLocalNetworkDataChange == true, error = OT_ERROR_INVALID_STATE);
 
-    SuccessOrExit(error = mDecoder.ReadIp6Address(routeConfig.mPrefix.mPrefix));
-    SuccessOrExit(error = mDecoder.ReadUint8(routeConfig.mPrefix.mLength));
+    SuccessOrExit(error = mDecoder.ReadIp6Prefix(routeConfig.mPrefix));
```

Comment:
How about if we fix the packed vs. unpacked issue by using a local `uint8_t prefixLength` variable that we pass to `ReadUint8()` instead and then assign the read value to `routeConfig.mPrefix.mLength`.

Even though adding a `ReadIp6Prefix()` makes the parsing code smaller, the intention behind `SpinelDecoder/Encoder` is for them to provide `Read/Write<Type>` methods mirroring the [primitive Spinel types](https://github.com/openthread/openthread/blob/master/doc/spinel-protocol-src/spinel-data-packing.md#primitive-types) and IP6 Prefix is not really a basic/primitive Spinel supported type. 

Thoughts?

Reply:
Sure, that was my initial idea, but i decided to implement ReadIp6Prefix method. I will make needed changes, thanks!

Reply:
Thanks.